### ISSUE TITLE
select.selectedOptions is not reactive

### ISSUE THREAD

**Issue by eight04** [jsdom/jsdom#2326]
### Basic info:

- **Node.js version:** 10.4.1
- **jsdom version:** 11.12.0

### Minimal reproduction case

```js
const {JSDOM} = require("jsdom");
const dom = new JSDOM(`<select multiple><option>1</option><option>2</option></select>`);
const select = dom.window.document.querySelector("select");
console.log(select.selectedOptions.length);
select.options[0].selected = true;
console.log(select.selectedOptions.length);
```
Result:
```
0
0
```
Here is the repro:
https://github.com/eight04/node-test/tree/jsdom-selected-options

### How does similar code behave in browsers?

(Link to a jsbin or similar suggested.)
http://jsbin.com/curiwopagu/edit?html,js,console,output
Result:
```
0
1
```

**Comment by domenic** [comment]
Great catch. [The code](https://github.com/jsdom/jsdom/blob/4d26c6773f011205d9c703cc5988e7c117efea31/lib/jsdom/living/nodes/HTMLSelectElement-impl.js#L128) seems reasonable, but you're definitely right that this is broken somehow. I wonder what's going on.

**Comment by Zirro** [comment]
Could it be a side-effect of the `[SameObject]` caching?

**Comment by Zirro** [comment]
Confirmed that the issue goes away without [SameObject]. There must be some nuance I'm missing with it, since its description clearly states that ["...when getting the value of the attribute on a given object, the same value must always be returned"](https://heycam.github.io/webidl/#SameObject).

Could it mean that the same HTMLCollection should be reused, but the nodes inside it should be replaced?

**Comment by domenic** [comment]
Yes, that's how it 's supposed to work. And the HTMLCollection constructor takes a filter function that gets reevaluated.

Maybe there's a cache that isn't being invalidated?

**Comment by Zirro** [comment]
I don't think there's any invalidation logic for https://github.com/jsdom/webidl2js/blob/master/lib/output/utils.js. Once the initial value has been obtained, it seems like it'll return it forever.

**Comment by domenic** [comment]
No, I mean for the contents of the HTMLCollection itself.

**Comment by Zirro** [comment]
Domenic figured out the cause! I'll submit a fix for this once I've written a test case.
